### ISSUE TITLE
xTitles Truncate Problem

### ISSUE THREAD

**Issue by bahadirarslan** [ReetuBok/MIMChart-Library#5]
Hi,

I implemented MIMChartLibrary to my project but i have a little but disturbing issue.

I am setting an array for x titles values shown below.
 clickDates =     (
        "03/11/2012",
        "04/11/2012",
        "05/11/2012",
        "06/11/2012",
        "07/11/2012",
        "08/11/2012"
    );

But chart couldn't draw all value on chart, only seems a part of value like "03/11/2" 
I tried to change xTitleSytle 1 to 4 but nothing changed.

Also changing height doesn't solve problem. 

If you can help me i will be appreciate.

**Comment by ReetuBok** [comment]
I will take a look at it.

**Comment by bahadirarslan** [comment]
Currently i am using xTitleStyle1 but other styles doesn't change the result, too.

Here is the code i am using..

[MIMColor InitColors];
            horizontalLinesProperties=nil;
            anchorPropertiesArray=nil;
             verticalLinesProperties=[NSDictionary dictionaryWithObject:[NSNumber numberWithFloat:15.0] forKey:@"gap"];
                         yValuesArray=[NSArray arrayWithArray:clickCounts];
             xValuesArray=[NSArray arrayWithArray:clickDates];
             xTitlesArray=[NSArray arrayWithArray:clickDates];

```
         lineGraph=[[MIMLineGraph alloc]initWithFrame:CGRectMake(0, mainViewY, 300, 300)];
         anchorPropertiesArray= [NSArray arrayWithObjects:[NSDictionary dictionaryWithObject:[NSNumber numberWithBool:YES] forKey:@"hideShadow"], nil];
         lineGraph.lineColorArray = [NSArray arrayWithObjects:[MIMColor GetMIMColorAtIndex:0],nil];            
         lineGraph.delegate=self;
         lineGraph.anchorTypeArray=[NSArray arrayWithObjects:[NSNumber numberWithInt:CIRCLE], nil];
         lineGraph.xTitleStyle=XTitleStyle1;
         [lineGraph drawMIMLineGraph];
```

**Comment by ReetuBok** [comment]
Hi,

I hope you are using version 2.23 of the library. 

verticalLinesProperties=[NSDictionary dictionaryWithObject:[NSNumber numberWithFloat:15.0] forKey:@"gap"];
I think this line is affecting your results. Let me quickly check. and see how I can fix it.
Try increasing that number from 15 to say 25.0
If you want to keep using 15.0 I will be able to take a look at it after some hours.

**Comment by ReetuBok** [comment]
Ok, Tested quickly with following code and I get following result:

verticalLinesProperties=[NSDictionary dictionaryWithObject:[NSNumber numberWithFloat:15.0] forKey:@"gap"];

yValuesArray=[[NSArray alloc]initWithObjects:@"10000",@"21000",@"24000",@"11000",@"5000",@"2000",@"9000",@"4000",@"10000",@"17000",@"15000",@"11000",nil];

```
        xValuesArray=[[NSArray alloc]initWithObjects:@"Jan",
                      @"Feb",
                      @"Mar",
                      @"Apr",
                      @"May",
                      @"Jun",
                      @"Jul",
                      @"Aug",
                      @"Sep",
                      @"Oct",
                      @"Nov",
                      @"Dec", nil];


        xTitlesArray=[[NSArray alloc]initWithObjects:@"01/01/2012",
                      @"02/01/2012",
                      @"03/01/2012",
                      @"04/01/2012",
                      @"05/01/2012",
                      @"06/01/2012",
                      @"07/01/2012",
                      @"08/01/2012",
                      @"09/01/2012",
                      @"10/01/2012",
                      @"11/01/2012",
                      @"12/01/2012", nil];


        mLineGraph=[[MIMLineGraph alloc]initWithFrame:CGRectMake(5, 30, myTableView.frame.size.width-20, myTableView.frame.size.width * 0.5)];
        mLineGraph.delegate=self;
        mLineGraph.tag=10+indexPath.row;

        //Set initial Y-Label as 0..
        mLineGraph.minimumLabelOnYIsZero=TRUE;


        mLineGraph.xTitleStyle=XTitleStyle1;

        //Set color for line graph
        MIMColorClass *c1=[MIMColorClass colorWithComponent:@"0,169,249"];
        mLineGraph.lineColorArray=[NSArray arrayWithObjects:c1, nil];


        [mLineGraph drawMIMLineGraph];
        [cell.contentView addSubview:mLineGraph];
```

In screenshot, the titles seem to be fine..don't go out of bound.. What is it you are doing different?

![Screen Shot 2013-02-18 at 4 13 09 PM](https://f.cloud.github.com/assets/194362/169031/3a75ff78-7a10-11e2-8768-81abb2c37c30.png)

**Comment by bahadirarslan** [comment]
I don't know why but changing gap value 15 to 25 solved my problem. Very thanks for your help..

You saved my night :) 

Also i wonder can we change text colors on x or y axis?

**Comment by ReetuBok** [comment]
But this is weird: I ran my test with gap 15 and it seems to work for me.
Which version of lib are you using latest 2.23 ?

Yes changing text color on x y is possible. I will tell you the example where it is done...I will have to find it myself first..

**Comment by ReetuBok** [comment]
WallChart Example 1.
You can find how to change text colors on x-y axis.

It is also mentioned on the documentation website:
http://mimchartlibrary.blogspot.com/2012/11/other-topics.html

Hope this helps.

**Comment by bahadirarslan** [comment]
Yes this is very weird because i noticed that sending more values cause this.
For example i am sending values below.

 clickDates =     (
        "10/10/2012",
        "11/10/2012",
        "12/10/2012",
        "13/10/2012",
        "14/10/2012",
        "15/10/2012",
        "16/10/2012",
        "17/10/2012",
        "25/01/2013",
        "29/01/2013",
        "08/02/2013",
        "17/02/2013"
    );

clickCounts =     (
        38,
        1,
        1,
        7,
        1,
        3,
        2,
        4,
        1,
        1,
        1,
        1
    );

And latest code i used

[MIMColor InitColors];
            horizontalLinesProperties=nil;
            anchorPropertiesArray=nil;
             verticalLinesProperties=[NSDictionary dictionaryWithObject:[NSNumber numberWithFloat:25.0] forKey:@"gap"];
                         yValuesArray=[NSArray arrayWithArray:clickCounts];
             xValuesArray=[NSArray arrayWithArray:clickDates];
             xTitlesArray=[NSArray arrayWithArray:clickDates];

```
         lineGraph=[[MIMLineGraph alloc]initWithFrame:CGRectMake(0, mainViewY, 300, 300)];
         anchorPropertiesArray= [NSArray arrayWithObjects:[NSDictionary dictionaryWithObject:[NSNumber numberWithBool:YES] forKey:@"hideShadow"], nil];
         lineGraph.lineColorArray = [NSArray arrayWithObjects:[MIMColor GetMIMColorAtIndex:0],nil];            
         lineGraph.delegate=self;
         lineGraph.anchorTypeArray=[NSArray arrayWithObjects:[NSNumber numberWithInt:CIRCLE], nil];
         lineGraph.xTitleStyle=XTitleStyle1;
         [lineGraph drawMIMLineGraph];
```

Here is the result
![IMG_0132](https://f.cloud.github.com/assets/130344/169100/06017f9a-7a12-11e2-8d1c-25efcec9096b.PNG)

I stucked here...

I recently downloaded codes from here and replaced library at project so i could use xTitleStyle enum, because it didn't exist on previous version that i have.

**Comment by ReetuBok** [comment]
Ok...I understand... I think I understand whats going on now.
The long line chart(when chart is drawn on the scrollview) .. I think the limit of gap in this particular case is 20.0 I guess from what I remember.I had restricted it for some reason .I will try to make it flexible. Thanks for your input.

**Comment by bahadirarslan** [comment]
Could you suggest me a work around for this to save my night :)

**Comment by ReetuBok** [comment]
Also I noticed, your y labels start from 1. You can set following variable to start it from 0 if you like.

 //Set initial Y-Label as 0..
mLineGraph.minimumLabelOnYIsZero=TRUE;

**Comment by ReetuBok** [comment]
Ok. Let me take a look for long line chart.

**Comment by bahadirarslan** [comment]
I changed y label starting and waiting your response about issue with prays :)

**Comment by ReetuBok** [comment]
Hmm..Interesting.. Now I am plotting the your values which you posted above on 300 px width.. and gap is set to 15px and mLineGraph.xTitleStyle=XTitleStyle1; and it is still looking fine. 

My feeling is that you are using a version which probably had this bug for x-y labels which you are experiencing.

**Comment by bahadirarslan** [comment]
Is there anyway to determine which version i am using?

**Comment by ReetuBok** [comment]
I would just say that download a fresh copy from github.

**Comment by bahadirarslan** [comment]
i am trying now.

**Comment by bahadirarslan** [comment]
I just redownloaded library and tried again but result is same. 
Let me take a look at simulator.

**Comment by bahadirarslan** [comment]
It is still same on simulator. decreasing y values changing width xtitles.

I couldn't understand why :(

**Comment by ReetuBok** [comment]
:) ...Now I don't understand why its behaving like that for you.... Is there a possibility that the dates have any newline characters in them ? like \n ?

**Comment by bahadirarslan** [comment]
I don't think so because i copied values from console, i think if there is \n we will see in results..

I am very appreciate for your help but this is very important for me and i stucked with it. Also i tried lots of other libraries and noticed that best one is this. 

Is there any way to change height o xtitlelabel on source code?Maybe i can find try-fail method.

**Comment by ReetuBok** [comment]
No... You can not change it as of now.

Most annoying thing is I am using the same code you have pasted above and it doesn't give me that kind of plotting !

I am going to send you the lib files on your email. Try removing the old files from your xcode project and add these files fresh. Somehow I have a feeling the old files are still in use in your project which is causing it.

**Comment by bahadirarslan** [comment]
I don't know how can i thank you really..

you may be right..

**Comment by ReetuBok** [comment]
Sent it to your gmail. All the best.
